src/Altinn.AccessManagement/Controllers/DelegationsController.cs

```diff
@@ -345,5 +346,44 @@ public async Task<ActionResult<List<MPDelegationExternal>>> GetMaskinportenSchem
                 return StatusCode(500);
             }
         }
+
+        /// <summary>
+        /// Endpoint for delegating maskinporten scheme resources between two parties
+        /// </summary>
+        /// <response code="400">Bad Request</response>
+        /// <response code="422">Unprocessable Entity</response>
+        /// <response code="500">Internal Server Error</response>
+        [HttpPost]
+        [Authorize(Policy = AuthzConstants.MPSystemResource_Write)]
+        [Route("accessmanagement/api/v1/{party}/delegations/maskinportenschema/")]
+        public async Task<ActionResult<DelegationOutputExternal>> MaskinportenDelegation([FromRoute] string party, [FromBody] DelegationInputExternal delegation)
+        {
+            int authenticatedUserId = AuthenticationHelper.GetUserId(HttpContext);
+            try
+            {
+                DelegationInput internalDelegation = _mapper.Map<DelegationInput>(delegation);
+                DelegationOutput response = await _delegation.MaskinportenDelegation(authenticatedUserId, party, internalDelegation);
+
+                if (response.Rights != null && response.Rights.Count > 0)
+                {
+                    DelegationOutputExternal delegationOutput = _mapper.Map<DelegationOutputExternal>(response);
+                    DelegationHelper.TryGetResourceFromAttributeMatch(response.Rights.First().Resource, out var _, out string resourceId, out var _, out var _);
+                    DelegationHelper.TryGetPartyIdFromAttributeMatch(internalDelegation.To, out int toPartyId);
+                    return Created(new Uri($"https://{Request.Host}/accessmanagement/api/v1/{party}/delegations/maskinportenschema/offered?to={toPartyId}&resourceId={resourceId}"), delegationOutput);
+                }
+
+                return new ObjectResult(ProblemDetailsFactory.CreateProblemDetails(HttpContext, (int)HttpStatusCode.UnprocessableEntity, detail: "Delegation could not be completed"));
+            }
+            catch (ValidationException valEx)
+            {
+                ModelState.AddModelError("Validation Error", valEx.Message);
+                return new ObjectResult(ProblemDetailsFactory.CreateValidationProblemDetails(HttpContext, ModelState));
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Internal exception occurred during maskinportenschema delegation");
+                return new ObjectResult(ProblemDetailsFactory.CreateProblemDetails(HttpContext, detail: ex.Message));
+            }
```

Comment:
Throwing and catching exceptions isn't the greatest pattern for handling flow of error information between Core and API.  It also currently lacks any errorcode/errorkey (and traceid for internal errors) which can be used to display a human readable error message to the user (with language support) by the frontend.

Reply:
Implementation changed from throwing and catching validation exceptions to setting error responses on the output model, which the controller must check and map to ValidationProblemDetails model when errors exists